adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -114,10 +114,18 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts map[str
 
 	apiLevelClusterNameProd := ""
 
+	// Enable retry for the routes if the upstream is within the same cluster
+	enableRSTRetry := false
+
 	// check API level production endpoints available
 	if mgwSwagger.GetProdEndpoints() != nil && len(mgwSwagger.GetProdEndpoints().Endpoints) > 0 {
 		apiLevelEndpointProd := mgwSwagger.GetProdEndpoints()
 		apiLevelbasePath = strings.TrimSuffix(apiLevelEndpointProd.Endpoints[0].Basepath, "/")
+
+		if apiLevelEndpointProd.Endpoints[0].URLType == "http" && strings.HasSuffix(apiLevelEndpointProd.Endpoints[0].Host, "svc.cluster.local") {
+			enableRSTRetry = true
+		}
```

Comment:
I forgot to mention one thing. Have we integrated Cilium into our CP Prod environment yet? I believe it's still in Dev, right? In our CP gateway, we use "svc.cluster.local" when incorporating our system APIs. Do we know if the same behavior we observed in DP is also occurring in CP in the Dev environment? If it's still in Dev, should we consider introducing a flag to disable this in the CP gateway until Cilium is also added to CP Prod?

Reply:
fixed in [gated RST retry with env var](https://github.com/wso2/product-microgateway/pull/3565/commits/f374037fc1e42959448fd37d56ab9a3984326fd5)